outreach/connector.go

```diff
@@ -33,19 +32,22 @@ func NewConnector(opts ...Option) (conn *Connector, outErr error) {
 		return nil, err
 	}
 
-	restApi, ok := providerInfo.GetOption(providerOptionRestApiURL)
-	if !ok {
-		return nil, fmt.Errorf("restAPIURL not set: %w", providers.ErrProviderOptionNotFound)
-	}
-
 	conn = &Connector{
 		Client: &common.JSONHTTPClient{
 			HTTPClient: params.Client.Caller,
 		},
-		BaseURL: restApi,
 	}
 
-	conn.Client.HTTPClient.Base = providerInfo.BaseURL
+	conn.setBaseURL(providerInfo.BaseURL)
 
 	return conn, nil
 }
+
+func (c *Connector) getURL(arg string) (*urlbuilder.URL, error) {
```

Comment:
I think this might be a little unintuitive for someone new - we should rename this to be a little more explicit (`getApiUrl` or something like that)